django/db/backends/base/features.py

```diff
@@ -308,3 +308,9 @@ def supports_transactions(self):
             count, = cursor.fetchone()
             cursor.execute('DROP TABLE ROLLBACK_TEST')
         return count == 0
+
+    def allows_group_by_selected_pks_on_model(self, model):
+        """ When allows_group_by_selected_pks is True, this method is used to
+        determine whether the particular model supports it.
+        Default: return model._meta.managed"""
+        return model._meta.managed
```

Comment:
This optimization is used only for backends with `allows_group_by_selected_pks` feature, we should use it here, e.g.
```python
def allows_group_by_selected_pks_on_model(self, model):
    if self.allows_group_by_selected_pks:
        return model._meta.managed
    return False
```

Reply:
That is handled in the compiler.py, this method is never called unless `allows_group_by_selected_pks` is True.

Reply:
I know but what it will use it somewhere else?

Reply:
We cannot return `True` if a backend doesn't support it at all.

Reply:
Just _unresolved_ this to keep it visible. Reading the code as it is now, I was "the `allows_group_by_selected_pks` is redundant — this method is never called in that case". 

If it's a pre-condition/guard, I'd usually separate it make obviously so:

```suggestion
def allows_group_by_selected_pks_on_model(self, model):
    if not self.allows_group_by_selected_pks:
        return False
    # Not a real comment: ...with the guard out of the way, actual method...
    return model._meta.managed
```

...but it's fine as it is.

Reply:
Agreed, fixed.